module/mempool/stdmap/eject.go

```diff
@@ -14,41 +14,123 @@ import (
-// from the entity set. It picks the first entity upon iteration, thus being the fastest
-// way to pick an entity to be evicted; at the same time, it conserves the random bias
-// of the Go map iteration.
-func EjectFakeRandom(entities map[flow.Identifier]flow.Entity) (flow.Identifier, flow.Entity) {
-	var entityID flow.Identifier
-	var entity flow.Entity
-	for entityID, entity = range entities {
-		break
-	}
-	return entityID, entity
-}
+type EjectFunc func(b *Backend) (flow.Identifier, flow.Entity, bool)
 
 // EjectTrueRandom relies on a random generator to pick a random entity to eject from the
 // entity set. It will, on average, iterate through half the entities of the set. However,
 // it provides us with a truly evenly distributed random selection.
-func EjectTrueRandom(entities map[flow.Identifier]flow.Entity) (flow.Identifier, flow.Entity) {
+func EjectTrueRandom(b *Backend) (flow.Identifier, flow.Entity, bool) {
 	var entityID flow.Identifier
 	var entity flow.Entity
+	var bFound bool = false
 	i := 0
-	n := rand.Intn(len(entities))
-	for entityID, entity = range entities {
+	n := rand.Intn(len(b.entities))
+	for entityID, entity = range b.entities {
 		if i == n {
+			bFound = true
 			break
 		}
 		i++
 	}
-	return entityID, entity
+	return entityID, entity, bFound
+}
+
+// EjectTrueRandomFast checks if the map size is beyond the
+// ideal size, and will iterate through them and eject unneeded
+// entries if that is the case.  Return values are unused, but
+// necessary because the LRUEjector is using the EjectTrueRandom
+func EjectTrueRandomFast(b *Backend) (flow.Identifier, flow.Entity, bool) {
+
+	// this is 64 for performance reasons - modulus 64 is a bit shift
+	const batchSize = 64
+
+	// 64 in batch plus a buffer of 64 to prevent boundary conditions from
+	// rounding errors.  The buffer is also here because the max index
+	// decreases as items are deleted
+	const threshold = batchSize * 2
+
+	var entities = b.entities
+
+	// dummy value
+	var retval flow.Identifier
+
+	// 'len' returns a uint.  This map is assumed to be < sizeof(uint)
+	mapSize := len(entities)
+
+	// this should never happen, and is just for a quick check
+	if b.ejectionTrigger > uint(mapSize) ||
+		threshold > uint(mapSize) {
+		return retval, nil, false
+	}
+
+	if (uint(mapSize) - b.ejectionTrigger) <= threshold {
+		// nothing to do, yet
+		return retval, nil, false
+	}
+
+	// generate 64 random numbers
+
+	// this array will store 64 indexes into the map
+	var mapIndexes [batchSize]int64
+
+	// starting point, create 64 random, sequentially increasing, values
+	var i uint = 0
+
+	for ; i < batchSize; i++ {
+		// get a random number (zero-based)
+		mapIndexes[i] = int64(rand.Intn(mapSize))
+	}
+
+	// this is the sort from golang/sort for small arrays
+	for i = 1; i < batchSize; i++ {
+		for j := i; mapIndexes[j] < mapIndexes[j-1]; j-- {
+			tmp := mapIndexes[j]
+			mapIndexes[j] = mapIndexes[j-1]
+			mapIndexes[j-1] = tmp
```

Comment:
```suggestion
			mapIndexes[j], mapIndexes[j-1] = mapIndexes[j-1], mapIndexes[j] 
```
You can do this as a one-liner with a [tuple assignment](https://golang.org/ref/spec#Assignments)

Reply:
I'm aiming for performance, so I'm avoiding tuples and their additional logic.

Reply:
Here's the godbolt with tuples: https://godbolt.org/z/b9o3cKv5q
Here's the godbolt without: https://godbolt.org/z/o586E61eh
Here's the diff:

```
--- output_auto 2021-07-23 08:57:38.000000000 -0400
+++ output_manual       2021-07-23 08:57:18.000000000 -0400
@@ -13,14 +13,14 @@
         FUNCDATA        $1, cccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccccc19b003b4032a46d3e8db29831f3fe(SB)
         FUNCDATA        $2, "".main.stkobj(SB)
         LEAQ    go.string."first"(SB), AX
-        MOVQ    AX, ""..autotmp_8+96(SP)
-        MOVQ    $5, ""..autotmp_8+104(SP)
+        MOVQ    AX, ""..autotmp_9+96(SP)
+        MOVQ    $5, ""..autotmp_9+104(SP)
         LEAQ    go.string."second"(SB), AX
-        MOVQ    AX, ""..autotmp_8+112(SP)
-        MOVQ    $6, ""..autotmp_8+120(SP)
+        MOVQ    AX, ""..autotmp_9+112(SP)
+        MOVQ    $6, ""..autotmp_9+120(SP)
         LEAQ    type.[2]string(SB), AX
         MOVQ    AX, (SP)
-        LEAQ    ""..autotmp_8+96(SP), AX
+        LEAQ    ""..autotmp_9+96(SP), AX
         MOVQ    AX, 8(SP)
         PCDATA  $1, $0
         CALL    runtime.convT2E(SB)
```

Reply:
Yup.  You're right.  Tuples result in fewer instructions.